diff --git a/src/libfetchers/github.cc b/src/libfetchers/github.cc
index a491d82a6..e12eaf87e 100644
--- a/src/libfetchers/github.cc
+++ b/src/libfetchers/github.cc
@@ -157,18 +157,31 @@ struct GitArchiveInputScheme : InputScheme
         return input;
     }
 
-    std::optional<std::string> getAccessToken(const std::string & host) const
+    std::optional<std::string> getAccessToken(const std::string & host, const std::string & url) const
     {
         auto tokens = fetchSettings.accessTokens.get();
+        std::string answer;
+        long unsigned int answer_match_len = 0;
+        if(! url.empty()){
+            for (auto & token : tokens) {
+                auto match_len = url.find(token.first);
+                if (match_len != std::string::npos && token.first.length() > answer_match_len) {
+                    answer = token.second;
+                    answer_match_len = token.first.length();
+                }
+            }
+            if (!answer.empty())
+                return answer;
+        }
         if (auto token = get(tokens, host))
             return *token;
         return {};
     }
 
-    Headers makeHeadersWithAuthTokens(const std::string & host) const
+    Headers makeHeadersWithAuthTokens(const std::string & host,const std::string & url) const
     {
         Headers headers;
-        auto accessToken = getAccessToken(host);
+        auto accessToken = getAccessToken(host, url);
         if (accessToken) {
             auto hdr = accessHeaderFromToken(*accessToken);
             if (hdr)
@@ -249,7 +262,10 @@ struct GitHubInputScheme : GitArchiveInputScheme
             : "https://%s/api/v3/repos/%s/%s/commits/%s",
             host, getStrAttr(input.attrs, "owner"), getStrAttr(input.attrs, "repo"), *input.getRef());
 
-        Headers headers = makeHeadersWithAuthTokens(host);
+        auto urlGen = fmt(
+            "%s/%s/%s",
+            host, getStrAttr(input.attrs, "owner"), getStrAttr(input.attrs, "repo"));
+        Headers headers = makeHeadersWithAuthTokens(host, urlGen);
 
         auto json = nlohmann::json::parse(
             readFile(
@@ -272,7 +288,10 @@ struct GitHubInputScheme : GitArchiveInputScheme
             host, getStrAttr(input.attrs, "owner"), getStrAttr(input.attrs, "repo"),
             input.getRev()->to_string(Base16, false));
 
-        Headers headers = makeHeadersWithAuthTokens(host);
+        auto urlGen = fmt(
+            "%s/%s/%s",
+            host, getStrAttr(input.attrs, "owner"), getStrAttr(input.attrs, "repo"));
+        Headers headers = makeHeadersWithAuthTokens(host, urlGen);
         return DownloadUrl { url, headers };
     }
 
@@ -316,7 +335,10 @@ struct GitLabInputScheme : GitArchiveInputScheme
         auto url = fmt("https://%s/api/v4/projects/%s%%2F%s/repository/commits?ref_name=%s",
             host, getStrAttr(input.attrs, "owner"), getStrAttr(input.attrs, "repo"), *input.getRef());
 
-        Headers headers = makeHeadersWithAuthTokens(host);
+        auto urlGen = fmt(
+            "%s/%s/%s",
+            host, getStrAttr(input.attrs, "owner"), getStrAttr(input.attrs, "repo"));
+        Headers headers = makeHeadersWithAuthTokens(host, urlGen);
 
         auto json = nlohmann::json::parse(
             readFile(
@@ -339,7 +361,10 @@ struct GitLabInputScheme : GitArchiveInputScheme
             host, getStrAttr(input.attrs, "owner"), getStrAttr(input.attrs, "repo"),
             input.getRev()->to_string(Base16, false));
 
-        Headers headers = makeHeadersWithAuthTokens(host);
+        auto urlGen = fmt(
+            "%s/%s/%s",
+            host, getStrAttr(input.attrs, "owner"), getStrAttr(input.attrs, "repo"));
+        Headers headers = makeHeadersWithAuthTokens(host, urlGen);
         return DownloadUrl { url, headers };
     }
 
@@ -379,7 +404,10 @@ struct SourceHutInputScheme : GitArchiveInputScheme
         auto base_url = fmt("https://%s/%s/%s",
             host, getStrAttr(input.attrs, "owner"), getStrAttr(input.attrs, "repo"));
 
-        Headers headers = makeHeadersWithAuthTokens(host);
+        auto urlGen = fmt(
+            "%s/%s/%s",
+            host, getStrAttr(input.attrs, "owner"), getStrAttr(input.attrs, "repo"));
+        Headers headers = makeHeadersWithAuthTokens(host, urlGen);
 
         std::string refUri;
         if (ref == "HEAD") {
@@ -426,7 +454,7 @@ struct SourceHutInputScheme : GitArchiveInputScheme
             host, getStrAttr(input.attrs, "owner"), getStrAttr(input.attrs, "repo"),
             input.getRev()->to_string(Base16, false));
 
-        Headers headers = makeHeadersWithAuthTokens(host);
+        Headers headers = makeHeadersWithAuthTokens(host, url);
         return DownloadUrl { url, headers };
     }
 
